packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -922,3 +1045,259 @@ class _CupertinoDragGestureController<T> {
     }
   }
 }
+
+class _CupertinoSheetScrollController extends ScrollController {
+  _CupertinoSheetScrollController({
+    required this.onDragStart,
+    required this.onDragUpdate,
+    required this.onDragEnd,
+    required this.sheetIsDraggedDown,
+  });
+
+  final void Function() onDragStart;
+  final void Function(double) onDragEnd;
+  final void Function(double) onDragUpdate;
+  final bool Function() sheetIsDraggedDown;
+
+  @override
+  _DraggableScrollableSheetScrollPosition createScrollPosition(
+    ScrollPhysics physics,
+    ScrollContext context,
+    ScrollPosition? oldPosition,
+  ) {
+    return _DraggableScrollableSheetScrollPosition(
+      physics: physics.applyTo(const AlwaysScrollableScrollPhysics()),
+      context: context,
+      oldPosition: oldPosition,
+      onDragStart: onDragStart,
+      onDragUpdate: onDragUpdate,
+      onDragEnd: onDragEnd,
+      sheetIsDraggedDown: sheetIsDraggedDown,
+    );
+  }
+}
+
+/// A scroll position that manages scroll activities for
```

Comment:
These docs looks like leftover copy paste from the widgets class. 
These should be updated to reflect the right context. To avoid confusion, it might be best to give this class a unique name.

Reply:
Also something for the doc - since this was copied from the widgets layer and then changed to suit the use case, the document should talk about what is different and why.